data/update.py

```diff
@@ -108,38 +112,83 @@ def run(options):
   if options.get("upload", False):
     print("[%s] Syncing scan data and database to S3." % the_date)
     print()
-    upload(the_date)
+    upload_s3(the_date)
     print()
     print("[%s] Scan data and database now in S3." % the_date)
 
   print("[%s] All done." % the_date)
 
 
 # Upload the scan + processed data to /live/ and /archive/ locations by date.
-def upload(date):
-  live_parents = "s3://%s/live/parents/" % (BUCKET_NAME)
-  live_subdomains = "s3://%s/live/subdomains/" % (BUCKET_NAME)
-  live_db = "s3://%s/live/db/" % (BUCKET_NAME)
-  archive_subdomains = "s3://%s/archive/%s/subdomains/" % (BUCKET_NAME, date)
-  archive_parents = "s3://%s/archive/%s/parents/" % (BUCKET_NAME, date)
-  archive_db = "s3://%s/archive/%s/db/" % (BUCKET_NAME, date)
-
+def upload_s3(date):
+  # Used for all operations.
   acl = "--acl=public-read"
 
-  shell_out(["aws", "s3", "sync", PARENTS_DATA, live_scanned, acl])
-  shell_out(["aws", "s3", "sync", SUBDOMAIN_DATA, live_subdomains, acl])
+  # Used when uploading to the live/ dir.
+  delete = "--delete"
+
+  live_parents = "s3://%s/live/parents/" % BUCKET_NAME
+  live_subdomains = "s3://%s/live/subdomains/" % BUCKET_NAME
+  live_db = "s3://%s/live/db/" % BUCKET_NAME
+
+  shell_out(["aws", "s3", "sync", PARENTS_DATA, live_parents, acl, delete])
+  shell_out(["aws", "s3", "sync", SUBDOMAIN_DATA, live_subdomains, acl, delete])
   shell_out(["aws", "s3", "cp", DB_DATA, live_db, acl])
 
-  # Ask S3 to do the copying, to save on time and bandwidth
-  shell_out(["aws", "s3", "sync", live_scanned, archive_scanned, acl])
-  shell_out(["aws", "s3", "sync", live_subdomains, archive_subdomains, acl])
-  shell_out(["aws", "s3", "sync", live_db, archive_db, acl])
+  # Then copy the entire live directory to a dated archive.
+  # Ask S3 to do the copying, to save on time and bandwidth.
+  live = "s3://%s/live/" % (BUCKET_NAME)
+  archive = "s3://%s/archive/%s/" % (BUCKET_NAME, date)
+  shell_out(["aws", "s3", "sync", live, archive, acl])
+
+
+# Makes use of the public URLs so that this can be run in a dev
+# environment that doesn't have write credentials to the bucket.
+def download_s3():
+
+  def download(src, dest):
+    # remote sources relative to bucket
+    url = "https://s3-us-gov-west-1.amazonaws.com/%s/%s" % (BUCKET_NAME, src)
```

Comment:
Note: the AWS region is hardcoded here. This should probably be moved out to `meta.yml` or something, alongside the bucket name.